frontend/src/app/VersionService.tsx

```diff
@@ -0,0 +1,24 @@
+export class VersionService {
+  /**
+   * fetch the commit SHA deployed with the static assets and compare it to the
+   * SHA encoded in the current build
+   */
+  static async getSHA(): Promise<string> {
+    if (process.env.NODE_ENV === "production") {
+      const result = await fetch(`${process.env.PUBLIC_URL}/static/commit.txt`);
```

Comment:
Does this do an Etag/if-modified-since check? Not that it makes much difference for a one line file, I just like it better that way.

Reply:
It appears that fetch's `default` cache control should do this under the hood, yes: 
- https://fetch.spec.whatwg.org/#concept-request-cache-mode
- https://fetch.spec.whatwg.org/#http-network-or-cache-fetch